spec/facility_spec.rb

```diff
@@ -17,10 +19,190 @@
+
+      @facility.add_service('Vehicle Registration')
+
+      expect(@facility.has_service?('Vehicle Registration')).to eq(true)
+
+    end
+  end
+
+  describe '#allow_service' do
+    it 'allows a service that has been added' do
+      @facility.add_service('Vehicle Registration')
+      @facility.has_service?('Vehicle Registration')
+
+      expect(@facility.allow_service('Vehicle Registration'){'Registered'}).to eq('Registered')     
+    end
+  end
+
+  describe '#allow_service' do
+    it 'doesnt allow a service that hasnt been added' do
+      expect(@facility.allow_service('Vehicle Registration'){'Registered'}).to eq(false)     
+    end
+  end
+
+  describe '#registered vehicles' do
+    it 'can list vehicles that have been registered at a facility' do
+      expect(@facility.registered_vehicles).to eq([])
+    end
+  end
+
+  describe '#collected fees' do
+    it 'shows that collected fees defaults at 0' do
+      expect(@facility.collected_fees).to eq(0)
+    end
+  end
+
+  describe '#register_vehicle' do
+    it 'can register a vehicle' do
+      @facility.add_service('Vehicle Registration')
+      @cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+
+      expect(@facility.register_vehicle(@cruz)).to eq([@cruz])
+    end
+  end
+
+  describe '#registered vehicles' do
+    it 'can add to registered vehicles array' do
+      @facility.add_service('Vehicle Registration')
+      @cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+      @facility.register_vehicle(@cruz)
+
+      expect(@facility.registered_vehicles).to eq([@cruz])
+    end
+  end
+
+  describe '#registration date' do
+    it 'can assign a registration date' do
+      @cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+      @facility.add_service('Vehicle Registration')
+      @facility.register_vehicle(@cruz)
+
+      expect(@cruz.registration_date).to eq(Date.today)
+    end
+  end
+
+  describe '#plate type' do
+    it 'plate type defaults to nil' do
+      @cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+
+      expect(@cruz.plate_type).to eq(nil)
+    end
+  end
+
+  describe '#plate type' do
+    it 'can assign plate type' do
+      @cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+      @facility.add_service('Vehicle Registration')
+      @facility.register_vehicle(@cruz)
+
+      expect(@cruz.plate_type).to eq(:regular)
+    end
+  end
+
+  describe '#collect fees' do
+    it 'can collect fees upon registration' do
+      @cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+      @facility.add_service('Vehicle Registration')
+      @facility.register_vehicle(@cruz)
+
+      expect(@facility.collected_fees).to eq(100)
```

Comment:
Yes, this is a good start, then we'd also want to register another vehicle and test that the collected fees attribute incremented how we expect.